.github/workflows/publish.yml

```diff
@@ -0,0 +1,35 @@
+name: publish
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        machine: [windows-latest, macos-latest, ubuntu-20.04]
+        architecture: [x64]
+        include:
+          - machine: windows-latest
+            platform: win
+          - machine: macos-latest
+            platform: osx
+          - machine: ubuntu-20.04
+            platform: linux
+    runs-on: ${{ matrix.machine }}
+    timeout-minutes: 5
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+      - name: publish
+        working-directory: src
+        run: dotnet publish -r ${{ matrix.platform }}-${{ matrix.architecture }} -c Release
```

Comment:
don't put `${{ ... }}` directly in scripts. remember what we learned about code injection in github actions?
while we do have control over the variables here, it's best to avoid this pattern entirely.

Reply:
also, since we're running on different OSes here, and different OSes have different default shell, we should explicitly specify `shell:` and pick one.